tests/nativeapp/test_artifacts.py

```diff
@@ -44,6 +46,512 @@ def dir_structure(path: Path, prefix="") -> List[str]:
     return parts
 
 
+@pytest.fixture
+def bundle_map():
```

Comment:
How about a test like the following, I'm curious what happens with these changes:

```
- src: src/snowpark
  dest: ./snowpark
- src: README.md
  dest: ./snowpark/
```

Reply:
Added, and it works (hopefully) exactly as you'd expect: it's allowed because in both cases the destination is a directory and there are no file collisions.

Reply:
I should expand: we don't currently have the ability to detect deep collisions (I added a skipped test that illustrates that). The new logic does not fix this, but I have some thoughts on how to make it possible. It will be a bigger change, but it would also help with some of our outstanding bugs.

Reply:
Can we add an assert that `README.md` doesn't end up in the `snowpark` source folder?

Reply:
Yeah, it was possible in the old logic to pollute the source directory. Added logic to explicitly disallow that, + an integ test to verify.